src/main/java/com/bandwidth/sqs/consumer/acknowledger/MessageAcknowledger.java

```diff
@@ -23,22 +24,30 @@
     private final SingleSubject<AckMode> ackModeSingle;
     private final CompletableSubject ackingComplete;
 
-    public MessageAcknowledger(SqsAsyncIoClient sqsClient, String queueUrl, String receiptId) {
+    private final MessagePublisher<T> messagePublisher;
+
+    public MessageAcknowledger(SqsAsyncIoClient sqsClient, String queueUrl, String receiptId,
+            MessagePublisher<T> messagePublisher) {
         this.sqsClient = sqsClient;
         this.queueUrl = queueUrl;
         this.receiptId = receiptId;
 
         this.ackModeSingle = SingleSubject.create();
         this.ackingComplete = CompletableSubject.create();
+        this.messagePublisher = messagePublisher;
     }
 
-    public MessageAcknowledger(MessageAcknowledger<?> delegate) {
+    /**
+     * Delegating constructor. Any ack method called on this instance will also call it on the delegate.
+     */
```

Comment:
Only this constructor delegates to the Acknowledger given, this has nothing to do with the entire class.